nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -187,6 +187,7 @@ async function findUsableCommitments(zkpPublicKey, ercAddress, tokenId, _value,
     );
   }
   commitmentsToUse.forEach(commitment => markNullified(commitment));
+  await Promise.all(commitmentsToUse);
```

Comment:
I don't know if this works as intended, `commitmentsToUse` isn't an array of promises so AFAIK this awaiting does nothing. `forEach` constructs are also a pattern to ignore returns.

We should move this to a `map` and return an array of promises that can then be awaited?

Reply:
Good spot!  Annoyingly my test case misses this one.